### PR TITLE
Deprecate PauliTable

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

- Deprecate `PauliTable`
- Deprecate `StabilizerTable.pauli`
- Make `StabilizerTable` a standalone (i.e. independent from PauliTable)

### Details and comments

TODO
- [x] Add Releasenote

### CODE REVIEW

**Comment by ikkoham** [comment]
Milestone: 0.19
Label: `Changelog: Deprecation`, `mod: quantum info`

**Review by chriseclectic — CHANGES_REQUESTED** [review]
Deprecating PauliTable is good, however keeping StabilizerTable  is avoiding the problem that the API for StabilizerTable (which is the same as PauliTable) is outdated and needs to go too since it differs quite drastically in some cases from PauliList.

There is no real need for a separate StabilierTable class when its functionality is a strict subset of functionality of PauliList. The only place StabilizerTable is used is in the Clifford class where the differences are StabilizerTable uses a Z_2 phase instead of Z_4, and stores the table as a single array instead of X and Z arrays separately. This can be addressed by directly adding `phase` and `array` setter and getter methods to the `Clifford` class that convert the PauliList to these formats as they were in the old StabilizerTable (stack X and Z, convert phase to/from Z_2). Then all the internal code in Clifford should just has to replace `obj.table.array` and `obj.table.phase` with `obj.array` and `obj.phase`.

The stabilizer and stabilizer methods can be updated to return the internal PauliLists, and the table method itself could either return the full PauliList or be deprecated.

**Comment by jakelishman** [comment]
@chriseclectic: I'd recommend against having non-trivial `@property` getter methods; if `x.phase` constructs a new array on every access, it's very very easy to introduce big performance issues without noticing.  If you're going to have data transformations on the class, it's probably best to put them in a "proper" function (so _don't_ use `@property`), so it's clear at the call site that it's an expensive call - calling `x.phase()` multiple times _looks_ wrong, which makes it safer from a performance perspective.

File: qiskit/quantum_info/operators/symplectic/stabilizer_table.py [hunk 1]
```diff
@@ -990,7 +1201,40 @@ def _to_matrix(pauli, phase, sparse=False):
             csr_matrix: if sparse=True.
         """
         # pylint: disable=arguments-differ
-        mat = PauliTable._to_matrix(pauli, sparse=sparse, real_valued=True)
+        def count1(i):
```
Comment (t-imamichi) [inline qiskit/quantum_info/operators/symplectic/stabilizer_table.py:1204]: I recommend to count 1 by `bin(i).count('1')`. It's simpler and faster. https://github.com/Qiskit/qiskit-terra/pull/5261#discussion_r531359343

Comment (ikkoham) [reply qiskit/quantum_info/operators/symplectic/stabilizer_table.py:1204]: Thank you. I use it.

**Comment by ikkoham** [comment]
I'll modify this PR after https://github.com/Qiskit/qiskit-terra/pull/7269. (So, don't merge this)

**Comment by ikkoham** [comment]
I'll deprecate PauliTable and StabilzierTable at the same time in 0.23.

**Pull request closed without merging**